roles/openshift_certificate_expiry/test/conftest.py

```diff
@@ -0,0 +1,130 @@
+        crypto.X509Extension(b'subjectKeyIdentifier', False, b'hash', subject=ca)
+    ])
+    ca.add_extensions([
+        crypto.X509Extension(b'authorityKeyIdentifier', False, b'keyid:always', issuer=ca)
+    ])
+    ca.sign(ca_key, 'sha256')
+    return ca
+
+
+@pytest.fixture(scope='session')
+def client_key():
+    key = crypto.PKey()
+    key.generate_key(crypto.TYPE_RSA, 4096)
+    return key
+
+
+@pytest.fixture(scope='session')
+def client_cert(ca_cert, ca_key, client_key):
+    cert = crypto.X509()
+    cert.set_serial_number(4)
+    cert.gmtime_adj_notBefore(0)
+    cert.gmtime_adj_notAfter(24 * 60 * 60)
+    cert.set_issuer(ca_cert.get_subject())
+    cert.set_pubkey(client_key)
+    cert.set_version(3)
+    cert.get_subject().commonName = 'host.example.com'
+    cert.add_extensions([
+        crypto.X509Extension(b'basicConstraints', True, b'CA:FALSE'),
+        crypto.X509Extension(b'keyUsage', True, b'digitalSignature, keyEncipherment'),
+        crypto.X509Extension(b'extendedKeyUsage', False, b'clientAuth'),
+    ])
+    cert.sign(ca_key, 'sha256')
+    return cert
+
+
+@pytest.fixture(scope='session')
+def client_cert_file(client_cert, ca_dir):
+    crt_file = ca_dir.join('client.crt')
+    with open(str(crt_file), 'wb') as f:
+        f.write(
+            crypto.dump_certificate(crypto.FILETYPE_PEM, client_cert)
+        )
+    return str(crt_file)
+
+
+@pytest.fixture(scope='session')
+def server_key():
+    key = crypto.PKey()
+    key.generate_key(crypto.TYPE_RSA, 4096)
+    return key
+
+
+@pytest.fixture(scope='session')
+def server_cert(ca_cert, ca_key, server_key):
+    hostname = 'host.example.com'
+    san_list = [
+        'DNS:{}'.format(hostname),
+        'DNS:kubernetes',
+        'DNS:kubernetes.default',
+        'DNS:kubernetes.default.svc',
+        'DNS:kubernetes.default.svc.cluster.local',
+        'DNS:openshift',
+        'DNS:openshift.default',
+        'DNS:openshift.default.svc',
+        'DNS:openshift.default.svc.cluster.local',
+        'DNS:10.0.0.1',
+        'DNS:192.168.0.1',
+        'IP:10.0.0.1',
+        'IP:192.168.0.1'
+    ]
+
+    cert = crypto.X509()
+    cert.set_serial_number(4)
+    cert.gmtime_adj_notBefore(0)
+    cert.gmtime_adj_notAfter(24 * 60 * 60)
+    cert.set_issuer(ca_cert.get_subject())
+    cert.set_pubkey(server_key)
+    cert.set_version(3)
+    cert.get_subject().commonName = hostname
+    cert.add_extensions([
+        crypto.X509Extension(b'basicConstraints', True, b'CA:FALSE'),
+        crypto.X509Extension(b'keyUsage', True, b'digitalSignature, keyEncipherment'),
+        crypto.X509Extension(b'extendedKeyUsage', False, b'serverAuth'),
+        crypto.X509Extension(b'subjectAltName', False, ', '.join(san_list).encode())
+    ])
+
+    cert.sign(ca_key, 'sha256')
+    return cert
+
+
+@pytest.fixture(scope='session')
+def server_cert_file(server_cert, ca_dir):
+    crt_file = ca_dir.join('server.crt')
+    with open(str(crt_file), 'wb') as f:
```

Comment:
Definitely need to clean it up, was trying to clean up the byte/str differences in py3 between pyopenssl and the fake class.